_source/user-guide/infrastructure-monitoring/custom-metrics.md

```diff
@@ -0,0 +1,192 @@
+contributors:
+  - shalper
+---
+
+You can send your application metrics to your Logz.io Infrastructure Monitoring account and benefit from all of the advantages of your tried and true dashboards.
+
+To do this, you'll need to send your application metrics to Logz.io in JSON format and stick to the formatting guidelines.
+
+##### Overview
+{:.no_toc}
+
+1. toc list
+{:toc}
+
+#### JSON format
+{:.no_toc}
+
+Our guiding principle is to use key-value pairs to define metrics and their dimensions.
+Here's the general template.
+
+  ```
+  {
+	"metrics": {
+		"my_first_metric.unit": number,
+		"my_second_metric.unit": number,
+		"my_third_metric.unit": number,
+		"my_fourtn_metrlc.unit": number
+	},
+	"dimensions": {
+		"dimension_l": "value",
+		"dimension_2": "value",
+		"dimension_3": "value",
+		"dimension_4": "value"
+	}
+  }
+  ```
+
+The reason being that we want to send _fewer and longer_ documents. 
+By grouping all possible metrics and dimensions into one document we maximize Elasticsearch's indexing power. In general, the _number_ of documents has a greater impact on Elasticsearch indexing power than document length.
+
+The best way to explain the guidelines is to look at a few examples. So let's get to it.
+
+<div class="tasklist">
+
+##### Single metric with dimensions
+
+In this example, we have one metric per document. We also have a metric field `name`, and its `value` is logged as a dimension.
+
+```
+{
+    "metric": {
+        "name": "refresh_page.duration"
+    },
+    "dimensions": {
+        "environment": "production",
+        "version": "12.05.01",
+        "unit": "milliseconds",
+        "value": 8
+    }
+}
+```
+
+This document should be rearranged so the two fields, `name` and `value` are a key-value pair. The field `unit` is made redundent if we add the units to the metric field name.
+
+Here's the result:
+
+```
+{
+    "metric": {
+        "refresh_page.duration.ms": 8
+    },
+    "dimensions": {
+        "environment": "production",
+        "version": "12.05.01"
+    }
+}
+```
+
+The metric is now named `refresh_page.duration.ms`. This makes it easier to query and visualize our metrics. An added advantage is that we can stack metrics with the same dimensions in the same document.
+
+##### Multiple metrics with the same dimensions
+
+Next, let's examine a list of metrics that share the same dimensions.
+
+By stating the measuring unit in the metric’s name, we prevent confusion, eliminate another dimension, and allow metric stacking in the same document. So it’s a win-win.
+
+```
+{
+	"metric": {
+		"RefreshPage.duration.ms": 8,
+		"AvgPageLoadDur.ms": 23,
+		"ClicksPerSession.count": 461,
+		"AvgMemPerOp": 1520000,
+		"AvgCPUPerOp": 0.000003
+	},
+	"dimensions": {
+		"SessionID": "f9f5d7ec-a354-5f5a-ae2d-d7f5973d07ad",
+		"Host_name": "118.2.45.162",
+		"Cluster": "Cluster-3",
+		"Environment": "Production",
+		"AppName": "Workshop",
+		"AppVersion": "12.05.01"
+	}
+}
+```
+
+##### Metrics vs. Dimensions
```

Comment:
This is a very important section. Might be worthy to put it at the top of the whole page - so users will have a clear understanding of what are metrics and especially what are dimensions.

I would also add a sentence about what actually are dimensions (because not all users might know this terminology) - saying that dimensions metadata about the metrics - from where it was sent, to which part it's relevant etc.)